File: pyround1/1431.拥有最多糖果的孩子.py
#
# @lc app=leetcode.cn id=1431 lang=python3
#
# [1431] 拥有最多糖果的孩子
#

# @lc code=start
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        res = []
        max_cad = max(candies)
        for i in candies:
            if i + extraCandies >= max_cad:
                res.append(True)
            else:
                res.append(False)
        return res
        
                
# @lc code=end

File: 1431.拥有最多糖果的孩子.cpp
/*
 * @lc app=leetcode.cn id=1431 lang=cpp
 *
 * [1431] 拥有最多糖果的孩子
 */

// @lc code=start
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool>res;
        int max_val = *max_element(candies.begin(), candies.end());
        for(auto num : candies){
            if(num + extraCandies >= max_val)
                res.push_back(true);
            else
            {
                res.push_back(false);
            }
        }
        return res;

    }
};
// @lc code=end

